sled-agent/src/services.rs

```diff
@@ -261,28 +311,32 @@ impl ServiceManager {
         debug!(log, "Creating new ServiceManager");
         let log = log.new(o!("component" => "ServiceManager"));
         let mgr = Self {
-            log: log.clone(),
-            config,
-            zones: Mutex::new(vec![]),
-            vnic_allocator: VnicAllocator::new("Service", etherstub),
-            physical_link_vnic_allocator: VnicAllocator::new(
-                "Public",
-                // NOTE: Right now, we only use a connection to one of the Chelsio
-                // links. Longer-term, when we we use OPTE, we'll be able to use both
-                // connections.
-                physical_link,
-            ),
-            underlay_vnic,
-            underlay_address,
-            rack_id,
-            ddmd_client: DdmAdminClient::new(log)?,
-            advertised_prefixes: Mutex::new(HashSet::new()),
+            inner: Arc::new(ServiceManagerInner {
+                log: log.clone(),
+                config,
+                // TODO: Load the switch zone if it already exists?
```

Comment:
Why would it already exist?  If the sled-agent crashed and restarted?  If so, I think we would want to shut down and recreate the zone, since we wouldn't know whether its services had been configured and started prior to the crash.  Totally fine to deal with that as a follow-on project.

Reply:
This is admittedly a fuzzy concern - *today*, it really *should not exist*, since the Sled Agent should be deleting all zones when the sled agent boots. https://github.com/oxidecomputer/omicron/issues/725 has more detail ; in general, we have mentioned the goal of "if the sled agent bounces, it would be nice to not kill the entire workload running on the sled".

I'll add a reference to this bug, so that if/when we remove the "zone deletion on sled agent startup", we can re-visit this spot.